### PR TITLE
Provided setup instructions for VSCode

### PR DESCRIPTION
**Pull request by icedjuro**
Added instructions for how to setup Visual Studio Code for using Godot. Instructions are made for Windows.

### CODE REVIEW

File: doc/setup_instructions.md [hunk 1]
```diff
@@ -84,12 +84,23 @@ desktop development featureset, that should have everything needed for
...
+
+Open up a new Godot Project. On the top, go to Editor -> Editor Settings. Scroll to the bottom, where you'll find Mono. Click on Editor and set External Editor to visual studio code. Click on Builds and set Build Tool to MSBuild (VS Build Tools).
+
+If you want to setup live debugging with Godot, on the top, go to Project -> Project Settings. Near the bottom, you will find Mono. Click on Debugger Agent. When you want to use the Debugger, turn the Wait for Debugger on. Set the Wait Timeout to how many milliseconds you want Godot to wait for visual studio code debugging, recommended to be set around 15000. Copy the port number and go to visual studio code. Make sure visual studio code is open to your Godot folder. Go to the debug tab and click on _create a launch.json file_. Select C# Mono. When the _launch.json_ file is opened, change the port number to the one you copied previously. Save the file and you can close it. On the Debug tab, switch Launch to Attach. Now, whenever you want to debug, make sure Wait for Debugger is on in Godot, run the project, and press the green arrow in the debugger in visual studio code.
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:103]: Could you add new lines to the text so that it is only around 80 columns wide

File: doc/setup_instructions.md [hunk 2]
```diff
@@ -235,19 +246,19 @@ code editor.
...
 command: `nuget restore` it should download the missing nuget
 packages. You may need to rerun this command when new package
 dependencies are added to Thrive. Note: if you use MonoDevelop it
 *might* automatically restore missing packages using nuget when
-compiling the game within MonoDevelop.
+compiling the game within MonoDevelop. 
 
 
-On Windows you should use Visual studio to restore the packages. To do
+On Windows you should use visual Studio to restore the packages. To do
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:257]: Why did you change this to be lower case? When referring to the software name uppercase should be used.

File: doc/setup_instructions.md [hunk 3]
```diff
@@ -235,19 +246,19 @@ code editor.
...
 this open `Thrive.sln` in the Thrive folder. The package restore might
 automatically happen if you compile the solution. If it doesn't please
 refer to this page on how to restore the nuget packages with visual
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:260]: `visual` here should be capitalized as well.

File: doc/setup_instructions.md [hunk 4]
```diff
@@ -84,12 +84,23 @@ desktop development featureset, that should have everything needed for
 C# development. Please be patient with this as this is going to take
 tens of minutes to install and take a lot of disk space.
 
-It might also be possible to make things work with Visual studio code
-as well, but there are no instructions for that. Note: that Godot
-requirements say that visual studio build tools are enough but they
-may not include nuget, which is needed, so if you go that route you
-may need to manually install nuget.
+It is also be possible to make things work with visual studio code
+as well, but the setup process is more complicated and more likely to not work. Note: that Godot requirements say that visual studio build tools are enough but they may not include nuget, which is needed, so if you go that route you may need to manually install nuget.
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:88]: Did you change anything here? To me it looks like you just made the text lines just longer.

File: doc/setup_instructions.md (hunk 1 above)
Comment (hhyyrylainen) [inline doc/setup_instructions.md:103]: To my eyes there are a bunch of sentences here where the structure could be made better and a few places I'm not sure if the text is good English.

Comment (icedjuro) [reply doc/setup_instructions.md:103]: Ok, will change that

File: doc/setup_instructions.md (hunk 2 above)
Comment (icedjuro) [reply doc/setup_instructions.md:257]: Noticed you referred to Visual Studio in lower case mostly, wanted to keep it consistent, will change that too.

File: doc/setup_instructions.md (hunk 4 above)
Comment (icedjuro) [reply doc/setup_instructions.md:88]: I don't remember changing anything there

File: doc/setup_instructions.md (hunk 1 above)
Comment (icedjuro) [reply doc/setup_instructions.md:103]: That's kind of unclear. I proof-read it to see if the grammar and spelling was OK. I'll check again to make sure the wording is OK. If you could tell me what you mean, like making it more concise or so, then I would know what to look for.

File: doc/setup_instructions.md [hunk 5]
```diff
@@ -1,6 +1,5 @@
 What's this?
 ============
-
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:3]: Please don't remove this blank line. All the other titles have a following blank line

File: doc/setup_instructions.md [hunk 6]
```diff
@@ -79,17 +78,57 @@ https://github.com/godotengine/godot-monodevelop-addin This is not
...
-It might also be possible to make things work with Visual studio code
+It might also be possible to make things work with Visual Studio Code
 as well, but there are no instructions for that. Note: that Godot
-requirements say that visual studio build tools are enough but they
+requirements say that Visual Studio build tools are enough but they
 may not include nuget, which is needed, so if you go that route you
 may need to manually install nuget.
 
+### Visual Studio Codeto
+To setup Visual Studio Code  work with Godot, you'll need
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:93]: Please add a blank line here. All the other titles have a following blank line

File: doc/setup_instructions.md [hunk 7]
```diff
@@ -79,17 +78,57 @@ https://github.com/godotengine/godot-monodevelop-addin This is not
...
 
-It might also be possible to make things work with Visual studio code
+It might also be possible to make things work with Visual Studio Code
 as well, but there are no instructions for that. Note: that Godot
-requirements say that visual studio build tools are enough but they
+requirements say that Visual Studio build tools are enough but they
 may not include nuget, which is needed, so if you go that route you
 may need to manually install nuget.
 
+### Visual Studio Codeto
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:92]: `Codeto` is a typo

File: doc/setup_instructions.md [hunk 8]
```diff
@@ -79,17 +78,57 @@ https://github.com/godotengine/godot-monodevelop-addin This is not
...
+To setup Visual Studio Code  work with Godot, you'll need
+Visual Studio Code, where you can install it here:
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:94]: I recommend wording like: You can install Visual Studio Code from here: LINK

File: doc/setup_instructions.md [hunk 9]
```diff
@@ -79,17 +78,57 @@ https://github.com/godotengine/godot-monodevelop-addin This is not
...
+https://code.visualstudio.com/
+
+Note: Setting up Visual Studio Code with Linux is possible,
+however it is recommended to use Monodevelp instead
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:98]: typo in MonoDevelop

File: doc/setup_instructions.md [hunk 10]
```diff
@@ -79,17 +78,57 @@ https://github.com/godotengine/godot-monodevelop-addin This is not
...
+
+Next, install Build Tools for Visual Studio here:
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:100]: "from here"

File: doc/setup_instructions.md [hunk 11]
```diff
@@ -79,17 +78,57 @@ https://github.com/godotengine/godot-monodevelop-addin This is not
...
+https://visualstudio.microsoft.com/downloads/?q=build+tools During the
+installation process, make sure MSBuild tools is listed under
+Installation details.
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:103]: `installation`

File: doc/setup_instructions.md [hunk 12]
```diff
@@ -79,17 +78,57 @@ https://github.com/godotengine/godot-monodevelop-addin This is not
...
+
+Go to https://dotnet.microsoft.com/download Under the .NET Core
+section, click on _Download .NET Core SDK_ and run the installer.
+Go back to the main download page and find
+_All .NET Framework Downloads_ Choose version 4.7 and select the Developer Pack.
+
+On Visual Studio Code, go to the Extensions tab. Get the extensions
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:110]: "Now open Visual Studio Code, and go to the ..."

File: doc/setup_instructions.md [hunk 13]
```diff
@@ -79,17 +78,57 @@ https://github.com/godotengine/godot-monodevelop-addin This is not
...
+_C#_, _Mono Debug_, and _godot-tools_.
+
+Open up a new Godot Project. On the top toolbar, go to Editor -> Editor
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:113]: `Open up a new Godot Project in the Godot editor`

File: doc/setup_instructions.md [hunk 14]
```diff
@@ -79,17 +78,57 @@ https://github.com/godotengine/godot-monodevelop-addin This is not
...
+Settings. Scroll down on the left window until you find Mono.
+Click on Editor and set External Editor to Visual Studio Code. Click on
+Builds and set Build Tool to MSBuild (VS Build Tools).
+
+If you want to setup live debugging with Godot, go to the top tooblar,
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:118]: `toolbar`

File: doc/setup_instructions.md [hunk 15]
```diff
@@ -79,17 +78,57 @@ https://github.com/godotengine/godot-monodevelop-addin This is not
...
+go to Project -> Project Settings. Scroll down on the left window until
+you find Mono. Click on Debugger Agent. When you want to use the debugger,
+turn the _Wait for Debugger_ setting on. Set the _Wait Timeout_ to how
+many milliseconds you want Godot to wait for your debugger to connect.
+Setting it to 15000 is recommended. Copy the port number and open up
+Visual Studio Code. Make sure Visual Studio Code has the Godot projcect
```
Comment (hhyyrylainen) [inline doc/setup_instructions.md:124]: `project`

File: doc/setup_instructions.md (hunk 10 above)
Comment (hhyyrylainen) [reply doc/setup_instructions.md:102]: Still an issue

File: doc/setup_instructions.md (hunk 11 above)
Comment (hhyyrylainen) [reply doc/setup_instructions.md:105]: capitalization still not fixed
